src/main/java/com/limechain/runtime/Runtime.java

```diff
@@ -1,34 +1,43 @@
 package com.limechain.runtime;
 
-import com.limechain.runtime.allocator.FreeingBumpHeapAllocator;
 import com.limechain.runtime.hostapi.WasmExports;
 import com.limechain.runtime.hostapi.dto.RuntimePointerSize;
 import com.limechain.runtime.version.RuntimeVersion;
-import lombok.Getter;
+import com.limechain.runtime.version.scale.RuntimeVersionReader;
+import com.limechain.utils.scale.ScaleUtils;
 import lombok.extern.java.Log;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 import org.wasmer.Instance;
-import org.wasmer.Memory;
 import org.wasmer.Module;
 
-import java.nio.ByteBuffer;
 import java.util.logging.Level;
 
-import static com.limechain.runtime.RuntimeBuilder.getImports;
-
-@Getter
 @Log
 public class Runtime {
-    private RuntimeVersion version;
-    private final Instance instance;
-    private final int heapPages;
-    private final FreeingBumpHeapAllocator allocator;
-
-    public Runtime(Module module, int heapPages) {
-        this.heapPages = heapPages;
-        this.instance = module.instantiate(getImports(module, this));
-        this.allocator = new FreeingBumpHeapAllocator(getHeapBase());
+    // TODO: Figure out whether we'll actually need this field
+    Module module; // used to open/close all of its own instances
```

Comment:
The `Module` field is left in here just for the sake of the `.close()` method. Initially, no such logic existed, but that's what I gathered from Gossamer; and since our `BlockTree` and its underlying `HashToMap` are modeled after Gossamer, I figured this might turn out to be what we want (closing the module as well). 

The main thing to consider is what this method actually does in `wasmer-java`, since there are significant differences in the public API and naming conventions between our `wasmer-java` and Gossamer's `wasmtime`. I haven't figured that out yet, so it's a TODO.

Reply:
`close` drops the Module pointer. Would be good to know whether that's necessary or simply a small improvement over the garbage collection.

Reply:
Delegated for the follow-up PR.

Reply:
This field remains for now. It seems, really, like it's not necessary, and rather a small improvement (optimisation for memory freeing), since `module.close()` essentially just drops the module pointer (and calls `nativeDrop` to drop the native object). In any case, bear this in mind if future considerations arise when refactoring - it's not the strictest of necessities.

No benchmarking has been done, however, so this is purely a conclusion from static inspections of the source code of `wasmer-java`.

Disclosed in 43db682b